### ISSUE TITLE
Handling MySQL column names with odd names (e.g. having `.` in it)

### ISSUE THREAD

**Issue by makesitgo** [mybatis/generator#315]
I am trying to run MBG against a database schema I do not have control over.  A few of the columns within a few of the tables have strange names: `park.key`, `span.first`, `span.last`, etc.

I have provided the necessary column overrides so that the Java code compiles fine.  However, when I try to test out the generated xml mappers, I run into errors like:

```
### Error updating database.  Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 'park.key' in 'field list'
### The error may involve com.lahman.core.mapper.BallparkMapper.insert-Inline
### The error occurred while setting parameters
### SQL: insert into Parks (park.key, park.name, park.alias, city, state, country)     values (?, ?, ?, ?, ?, ?)
### Cause: com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 'park.key' in 'field list'
; bad SQL grammar []; nested exception is com.mysql.jdbc.exceptions.jdbc4.MySQLSyntaxErrorException: Unknown column 'park.key' in 'field list'
```

I'm wondering if this might just be a limitation of the framework?  The `ref.field` pattern is used heavily in the xml mappers for denoting actual field values.  Is it possible, in the xml mappers, to reference a column which has a `.` in its name?  Here is a sample xml snippet of what's causing me issues:

```xml
<mapper namespace="com.lahman.core.mapper.BallparkMapper">
  <resultMap id="BaseResultMap" type="com.lahman.types.ballpark.Ballpark">
    <constructor>
      <idArg column="park.key" javaType="java.lang.String" jdbcType="VARCHAR"/>
      <arg column="park.name" javaType="java.lang.String" jdbcType="VARCHAR"/>
      <arg column="park.alias" javaType="java.lang.String" jdbcType="VARCHAR"/>
      <arg column="city" javaType="java.lang.String" jdbcType="VARCHAR"/>
      <arg column="state" javaType="java.lang.String" jdbcType="VARCHAR"/>
      <arg column="country" javaType="java.lang.String" jdbcType="VARCHAR"/>
    </constructor>
  </resultMap>
  <!-- more mapper content here -->
  <insert id="insert" parameterType="com.lahman.types.ballpark.Ballpark">
    insert into Parks (park.key, park.name, park.alias, city, state, country)
    values (#{key,jdbcType=VARCHAR}, #{name,jdbcType=VARCHAR}, #{alias,jdbcType=VARCHAR}, #{city,jdbcType=VARCHAR}, #{state,jdbcType=VARCHAR}, #{country,jdbcType=VARCHAR})
  </insert>
```

Just for completeness, here's a sampling of the Ballpark type that's being referenced above:

```java
public class Ballpark {

    private String key;
    private String name;
    private String alias;
    private String city;
    private String state;
    private String country;

    protected Ballpark() {
        // do nothing
    }

    public Ballpark( final String key, final String name, final String alias, final String city, final String state, final String country ) {
        this.key = key;
        this.name = name;
        this.alias = alias;
        this.city = city;
        this.state = state;
        this.country = country;
    }

  // getters/setters
}
```

**Comment by jeffgbutler** [comment]
Try setting `delimitedColumnName=true` on your column overrides.

**Comment by makesitgo** [comment]
Thanks for the suggestion ... I always wondered what that setting did ... and it nearly got me there.  After setting that flag, I started seeing those irregular names appear wrapped in quotations in my mappers' sql statements.  This still didn't work for MySQL, though it was for different reasons: invalid syntax.  After some manual tweaking/testing, I discovered that wrapping the column name in backticks in fact did solve my problem.

Is there a way within MBG to configure using backticks instead quotes for the `delimitedColumnName` setting?
